qiskit/primitives/containers/bindings_array.py

```diff
@@ -351,7 +417,11 @@ def examine_array(*possible_shapes):
 
 
 def _format_key(key: tuple[Parameter | str, ...]):
-    return tuple(k.name if isinstance(k, Parameter) else k for k in key)
+    return tuple(map(_param_name, key))
+
+
+def _param_name(param: Parameter | str):
```

Comment:
minor but maybe useful type hint because vscode shows `(str | Any)` of this return type.
```suggestion
def _param_name(param: Parameter | str) -> str:
```

Reply:
Since it's very minor, I merge this PR anyways.